hudi-aws/src/test/java/org/apache/hudi/aws/sync/ITTestGluePartitionPushdown.java

```diff
@@ -131,8 +151,40 @@ public void testEmptyPartitionShouldReturnEmpty() {
 
   @Test
   public void testPresentPartitionShouldReturnIt() throws ExecutionException, InterruptedException {
-    createPartitions("1", "b'ar");
+    List<String> partitions = createPartitions(1, 2, true);
     Assertions.assertEquals(1, glueSync.getPartitionsFromList(TABLE_NAME,
-            Arrays.asList("1/b'ar", "2/foo", "1/b''ar")).size());
+            Arrays.asList(partitions.get(0), "2/foo", "1/b''ar")).size());
```

Comment:
not sure how this test can work. The createPartition likely doesn't create what's in the assertion anymore.

Reply:
this test does the following:
1. create 1 partition
2. Request 3 partitions(1 created at first step and 2 random ones - copied from previous version).
3. Assert only real one returned

Reply:
:+1:

Reply:
for clarity, maybe : 
```suggestion

            Arrays.asList(partitions.get(0), "1/non-existing", "2/other-non-existing")).size());
```